### PR TITLE
feat: new methods for RobustRandom - getting random item from collections, getting random sub-set of items.

### PR DESCRIPTION
**Pull request by Fildrance**
Refined xml-doc for IRobustToolbox.
Added new methods for working with collections:
```
    T GetItem<T>(IList<T> collection);

    T GetItem<T>(ValueList<T> list);

    T GetItem<T>(Span<T> span);

    IReadOnlyCollection<T> GetItems<T>(IList<T> collection, int count, bool allowDuplicates = true);

    ValueList<T> GetItems<T>(ValueList<T> collection, int count, bool allowDuplicates = true);

    Span<T> GetItems<T>(Span<T> collection, int count, bool allowDuplicates = true);
```
Moved `Shuffle ` code from interface to implementation (as `GetItems` can benefit from its usage).

Main scenarios for new methods:
* Need to pick item random from list of items
* Need to pick several random items from list, only unique
* Need to pick items from list several times, duplicates allowed.

### CODE REVIEW

File: Robust.Shared/Random/RobustRandom.cs [hunk 1]
```diff
@@ -1,58 +1,243 @@
...
+        return _random.NextDouble();
+    }
+
+    public void NextBytes(byte[] buffer)
+    {
+        _random.NextBytes(buffer);
+    }
+
+    /// <inheritdoc />
+    public void Shuffle<T>(IList<T> list)
```
Comment (Fildrance) [inline Robust.Shared/Random/RobustRandom.cs:62]: this is transfered from interface to be used inside implementation, as implementation have no access to interface-implemented methods.
EVERYTHING BEFORE THIS ROW IS JUST CHANGE TO FILE-SCOPED-NAMESPACE

**Comment by Fildrance** [comment]
I'm sorry about chaning namespace and making code-review for this thing harder than it should be. I can revert it if it helps.

**Comment by EmoGarbage404** [comment]
Isnt this just Pick

**Comment by Fildrance** [comment]
If everything is ok i overall i'll try add some unit(-like) tests!

**Comment by Fildrance** [comment]
Pick is in extensions, i see, i will remove thouse 3 methods from interface then. But 'pick' N or 'pick N unique' is not there sadly.

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Make separate commits for style changes, doc comment additions and actual API changes. I'm not reviewing this mega commit.

**Comment by Fildrance** [comment]
> Make separate commits for style changes, doc comment additions and actual API changes. I'm not reviewing this mega commit.

fair enough. i'll make it in other branch so i'm closing this one.

**Pull request closed without merging**